137 UI fixes

KlamChowder1 opened the pull request:
bunch of UI updates 
- added tags to container card in flowchart
- made horizontal nodes 
![image](https://user-images.githubusercontent.com/24284083/112092570-02c6dc00-8b55-11eb-987b-2c2d5c96ab0e.png)

- added tags for default containers - couldn't reproduce the 'From: user' chip overlap bug with the tag chips... looks fine on my end 
![image](https://user-images.githubusercontent.com/24284083/112092637-27bb4f00-8b55-11eb-9c7f-31a7e7f95ed2.png)
![image](https://user-images.githubusercontent.com/24284083/112092627-22f69b00-8b55-11eb-9ea7-db28fcfb2484.png)

- added Qurit logo at bottom of drawer
![image](https://user-images.githubusercontent.com/24284083/112092792-77017f80-8b55-11eb-9286-cef87e0fe0cd.png)

- changed Container page to mirror Pipeline page (except no share Containers because I don't think other users should be able to see that) 
![container-page](https://user-images.githubusercontent.com/24284083/112092933-d3649f00-8b55-11eb-948a-649433708d89.gif)

- no more pop-up confirmation if nothing was modified (modified add/delete/move nodes, add/delete link, change destination, set conditions)
![unsaved-changes](https://user-images.githubusercontent.com/24284083/112093145-263e5680-8b56-11eb-8868-0cfc7912a13b.gif)

- container card drawer on flowchart filters by tags (doesn't need to exact match anymore) + filter by container owner
![filiter_contianer_cards](https://user-images.githubusercontent.com/24284083/112093309-6bfb1f00-8b56-11eb-8ba3-d5caa6ec0349.gif)

- having some trouble with the initial scene set thing so that pipeline nodes aren't out of view 🤔 will continue looking into it 
- TODO: remove redundant code/refactor, give Vue components their names + propTypes, add comments to code and stuff probably

awtkns commented:
@KlamChowder1 this looks awesome!

awtkns commented on frontend/components/container/ContainerForm.vue:
```diff
@@ -3,7 +3,7 @@
       <v-toolbar-title>
         <b
-          >{{ !!containerToEdit ? 'Edit your Container' : 'Add a Container' }}
+          >{{ !!isEditing ? 'Edit your Container' : 'Add a Container' }}
```
```js
<v-toolbar-title
v-text="isEditing ? 'Edit your Container' : 'Add a Container'"
class="font-weight-bold
/>
```

awtkns commented on frontend/components/container/ContainerForm.vue:
```diff
@@ -63,7 +63,7 @@
           text
         >
-          {{ !!containerToEdit ? 'Save Edits' : 'Add Container' }}
+          {{ !!isEditing ? 'Save Edits' : 'Add Container' }}
```
we can make this a v-icon-btn

awtkns commented on frontend/components/container/ContainerForm.vue:
```diff
@@ -104,7 +107,7 @@ export default {
     // disables button if no name or dockerfile for new container
     isDisabled: function() {
-      return !!this.containerToEdit
+      return !!this.isEditing
```
why do we need !! on a boolean?

awtkns commented on frontend/components/container/ContainerList.vue:
```diff
@@ -15,6 +15,11 @@
         hide-details
         solo
       />
+      <v-btn icon>
+        <v-icon @click="addContainerDialog" large color="#373740"
+          >mdi-plus</v-icon
+        >
+      </v-btn>
```
Can make this a v icon btn

awtkns commented on frontend/components/nav/NavDrawer.vue:
```diff
@@ -63,6 +63,16 @@
         </v-tooltip>
       </v-col>
       <v-col cols="12" class="text-center pt-0">{{ $auth.user.name }}</v-col>
+      <v-col>
+        <a href="https://qurit.ca/">
+          <v-img
+            :src="require('@/static/qurit-logo.png')"
+            max-width="100px"
+            contain
+            id="qurit"
+          />
+        </a>
+      </v-col>
```
Can we add a bit of  bottom padding?

KlamChowder1 commented:
- added a toolbar thing for add pipeline form for consistency with add container form
![image](https://user-images.githubusercontent.com/24284083/112745219-93d5f280-8f5b-11eb-99d9-40961531a6ae.png)
- fixed container and pipeline page
![image](https://user-images.githubusercontent.com/24284083/112745336-676ea600-8f5c-11eb-8c4c-c6b02c06af47.png)
- adjusted chips in pipeline builder for shared containers
![image](https://user-images.githubusercontent.com/24284083/112745352-8a995580-8f5c-11eb-802f-688e1de143da.png)

awtkns commented:
@KlamChowder1 would you be able to add a form to the admin page so that admins an create a local user?

KlamChowder1 commented:
- added vuex store for users (not sure if we want to add a delete user route?)
- added create local user form (pretty basic not sure if we want more)
![image](https://user-images.githubusercontent.com/24284083/112805528-89365e80-902a-11eb-9b50-a2113b93eef5.png)
- fixed up all console warnings now 
- added types to all props

KlamChowder1 commented on frontend/components/container/ContainerForm.vue:
```diff
@@ -83,7 +84,7 @@ export default {
   data() {
     return {
-      file: '',
+      file: [],
```
had to do this to get rid of the console warning 
https://stackoverflow.com/questions/60332960/vue-vuetify-invalid-prop-custom-validator-check-failed-for-prop-value-found

awtkns commented on backend/api/models/user.py:
```diff
@@ -16,8 +16,8 @@ class User(Base):
     is_admin = Column(Boolean, default=False)
-    first_seen = Column(DateTime, default=datetime.utcnow)
-    last_seen = Column(DateTime, default=datetime.utcnow)
+    first_seen = Column(DateTime, default=datetime.now())
+    last_seen = Column(DateTime, default=datetime.now())
```
@KlamChowder1 Is there any reason you changed this? Pretty this will just return the date the server was started as the default date. Other than that everything looks good.

```suggestion
    first_seen = Column(DateTime, default=datetime.now)
    last_seen = Column(DateTime, default=datetime.now)
```

KlamChowder1 replied:
@awtkns yea the server would show the date but in UTC, so I just did `now() ` so that the datetime displayed would be in local time so users can read it more easily

awtkns commented:
@KlamChowder1 I also noticed there is a issue (zindex?) with the output cards that makes it hard to connect the pipelines

KlamChowder1 commented:
- fixed z-index for node port
![image](https://user-images.githubusercontent.com/24284083/112955083-e9dd9e00-90f3-11eb-8152-3f47f35d2483.png)
- added password visibility toggle
![image](https://user-images.githubusercontent.com/24284083/112955142-f82bba00-90f3-11eb-8fe3-40f638b47b9c.png)

awtkns reviewed (approved):
LGTM!!
